From ab363f10402ea531849b7de18d5180b47d9a807f Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Tue, 1 Oct 2024 14:10:35 +0200
Subject: [PATCH 1244/1695] clk: stm32mp25: calibration: fix HSI deviation
 parameter from DT

HSI deviation parameter from DT was not take in count (always default
value).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 5adce3f0f622502390960426e1c1b88b13580a49 clk: stm32mp25: calibration: fix HSI deviation parameter from DT]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I75f97d760a5f9d8b217d45712b4ea55882399e46
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/408052
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/clk/clk-stm32mp2-cal.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/clk/clk-stm32mp2-cal.c b/core/drivers/clk/clk-stm32mp2-cal.c
index df7c40641..f34a0356b 100644
--- a/core/drivers/clk/clk-stm32mp2-cal.c
+++ b/core/drivers/clk/clk-stm32mp2-cal.c
@@ -419,7 +419,7 @@ static TEE_Result stm32mp2_clk_init_hsi_cal(struct stm32_calib_pdata *pdata)
 	/* Set HSI clock count deviation value */
 	assert(pdata->hsi_ref <= RCC_HSIFMONCR_HSIDEV_MASK);
 	io_clrsetbits32(address, RCC_HSIFMONCR_HSIDEV_MASK,
-			pdata->hsi_ref << RCC_HSIFMONCR_HSIDEV_SHIFT);
+			pdata->hsi_dev << RCC_HSIFMONCR_HSIDEV_SHIFT);
 
 	/* Set HSI clock count reference value */
 	io_clrsetbits32(address, RCC_HSIFMONCR_HSIREF_MASK,
-- 
2.47.2

